src/TextInputWidget.cc

```diff
@@ -101,6 +111,51 @@ FilteredTextEdit::keyPressEvent(QKeyEvent *event)
         }
 }
 
+bool
+FilteredTextEdit::canInsertFromMimeData(const QMimeData *source) const
+{
+        return (source->hasImage() || source->hasText() ||
+                QTextEdit::canInsertFromMimeData(source));
+}
+
+void
+FilteredTextEdit::insertFromMimeData(const QMimeData *source)
+{
+        if (source->hasImage()) {
+                const QImage image = qvariant_cast<QImage>(source->imageData());
+
+                const auto formats = source->formats();
+                const auto idx     = formats.indexOf(
+                  QRegularExpression{"image/\\w+", QRegularExpression::CaseInsensitiveOption});
```

Comment:
Is there a reason to be more selective than `image/.*`?

Reply:
Not necessarily. I just find it more reassuring than one that gives `image/` as a potential match. I do not mind changing it if you prefer.

Reply:
Mainly I'm leery of inadvertently filtering out future valid image types. It's not immediately obvious to me that Qt's interpretation of `\w` is a superset of valid media type characters. I wouldn't object to `image/.+` either, but if something's generating clipboard contents with garbage mime-types anything we do is going to be best-effort anyway.

Reply:
Oops I forgot to address this one. I will make the change locally, and add it to the next set of changes if there is one.